File: genna/smbase/smbaseJava/__smbase/EventHandle.java
package __smbase;

import java.util.LinkedList;

public class EventHandle {

    private boolean isEnabled;
    private LinkedList<Transition> transitionSequence;
    private State target;

    public EventHandle(LinkedList<Transition> ts, State t){
	transitionSequence = ts;
	target = t;
    }

    public LinkedList<Transition> getTransitionSequence(){
	return transitionSequence;
    }

    public State getTarget(){
	return target;
    }
}

File: genna/smbase/smbaseCpp/__smbase/EventHandle.cpp
#include "EventHandle.h"

namespace __smbase{

  EventHandle::EventHandle(std::vector<Transition *> ts, State *t){
    transitionSequence = ts;
    target = t;
  }

  std::vector<Transition *> EventHandle::getTransitionSequence(){
    return transitionSequence;
  }

  State *EventHandle::getTarget(){
    return target;
  }
}
